src/LaunchDarkly.ServerSdk/Internal/BigSegments/BigSegmentStoreWrapper.cs

```diff
@@ -71,19 +71,17 @@ public void Dispose()
         /// <returns>the query result</returns>
         internal BigSegmentsQueryResult GetUserMembership(string userKey)
         {
-            var ret = new BigSegmentsQueryResult();
             try
             {
-                ret.Membership = _cache.Get(userKey); // loads value from store via QueryMembership if not already cached
-                ret.Status = GetStatus().Stale ? BigSegmentsStatus.Stale : BigSegmentsStatus.Healthy;
+                IMembership membership = _cache.Get(userKey); // loads value from store via QueryMembership if not already cached
+                BigSegmentsStatus status = GetStatus().Stale ? BigSegmentsStatus.Stale : BigSegmentsStatus.Healthy;
+                return new BigSegmentsQueryResult(membership, status);
             }
             catch (Exception e)
             {
                 LogHelpers.LogException(_logger, "Big segment store returned error", e);
-                ret.Membership = null;
-                ret.Status = BigSegmentsStatus.StoreError;
+                return new BigSegmentsQueryResult(membership: null, BigSegmentsStatus.StoreError);
             }
-            return ret;
```

Comment:
Properties never mutated after returning, so it should actually have been declared immutable with `readonly struct`.